core-rust/core-api-server/src/core_api/handlers/stream_transactions.rs

```diff
@@ -47,22 +47,30 @@ pub(crate) async fn handle_stream_transactions(
 
     let max_state_version = database.max_state_version();
 
-    let txns = database.get_committed_transaction_bundles(
+    let transactions = database.get_committed_transaction_bundles(
         from_state_version,
         limit.try_into().expect("limit out of usize bounds"),
     );
 
-    let api_txns = txns
-        .into_iter()
-        .map(|(ledger_transaction, receipt, identifiers)| {
-            Ok(to_api_committed_transaction(
-                &mapping_context,
-                ledger_transaction,
-                receipt,
-                identifiers,
-            )?)
-        })
-        .collect::<Result<Vec<models::CommittedTransaction>, ResponseError<()>>>()?;
+    // Reserve enough for the "header" fields
+    let mut current_total_size = 2048;
```

Comment:
That looks super fragile... I'm quite sure it will break at some point.

Reply:
I agree. Let me try another good-enough approximation.

Reply:
Given that we don't really have a hard-and-fast limit, more a loose cap to try and avoid massive responses / OOM in node and clients, I think we could also just make this an estimate and it's fine.